### DISCUSSION TITLE
Mutliple HTTP methods for single endpoint?

### DISCUSSION THREAD

**Discussion by azzamsa** [poem-web/poem#229] [Q&A] [answered]
Hi.

I am porting this code from plain Poem to Poem-OpenApi.

```rust
    let app = Route::new()
        .at("/foo/*path", get(routes::risetai).post(routes::risetai).delete(routes::risetai))
```

```rust
#[OpenApi]
impl ProxyHandler {
    #[oai(path = "/foo/*path", method = "get")]
    pub async fn proxy(
        &self,
```

How to use multiple HTTP methods in one single endpoint?
Something like:

```diff
#[OpenApi]
impl ProxyHandler {
-    #[oai(path = "/foo/*path", method = "get")]
+    #[oai(path = "/foo/*path", methods = ["get", "post", "delete")]
    pub async fn proxy(
        &self,
```

I can't find one single endpoint used by many HTTP methods in examples:
- https://github.com/poem-web/poem/blob/f2bc9ca1ef540da48b2cdaab30466ee723058a15/examples/openapi/todos/src/main.rs#L43
- And in other examples

Thanks in advance.

**Comment by sunli829** [comment, marked as answer]
Specifying multiple methods for the same endpoint is not yet supported, I can add it.

Also, the OpenAPI specification does not support wildcard path matching, so `Poem-openapi` does not support it either.

**Reply by azzamsa** [reply]
> Also, the OpenAPI specification does not support wildcard path matching, so Poem-openapi does not support it either.

But, I can use `/foo/bar-path` and `/foo/baz-path` just fine in Poem-openapi. Is this means I can use wildcard path matching in Poem-openapi but it can't generate the docs?

**Reply by sunli829** [reply]
Yes, you can use it, but `swagger ui` doesn't recognize it.

**Reply by azzamsa** [reply]
This is the output of the Swagger UI. I think this is okay for me.

![image](https://user-images.githubusercontent.com/17734314/157784148-c3f20b5c-2fde-4805-aa54-4a9207f352b7.png)

Regarding the `multiple methods`. Is there any other approach I could take while I wait for it to be implemented in `poem-openapi`?

**Reply by sunli829** [reply]
I've done it and will release a new version later. 😁

https://github.com/poem-web/poem/blob/c44dcd7f0a140078212e0a76fef019bfb15f2c33/poem-openapi/tests/api.rs#L653

**Reply by sunli829** [reply]
Released in `v1.3.14`

**Reply by azzamsa** [reply]
Thank you for making a release. Using `poem = { git = "https://github.com/poem-web/poem", rev = "c44dcd7", features = ["test"] }` gives me lot of errors.

**Comment by duskmoon314** [comment]
> Specifying multiple methods for the same endpoint is not yet supported, I can add it.

Any news on this? I am looking for this feature and it seems still not yet supported.

**Reply by duskmoon314** [reply]
I currently use the way below:

```rust
async fn inner(&self) -> Res {...}

#[oai(path = "/xxx", method = "get")]
async fn get(&self) -> Res {
  self.inner().await
}
```

**Reply by sunli829** [reply]
This is already supported.

https://github.com/poem-web/poem/blob/3f10656f7b78f00e5b16dcd232746eba0c73e3d3/poem-openapi/tests/api.rs#L653
